spec/src/main/asciidoc/architecture.asciidoc

```diff
@@ -268,17 +650,349 @@ public PublisherBuilder<IncomingMessage, OutgoingMessage> processMessages() {
+|Default Acknoledgement Strategy
+|Supported Strategies
+
+|
+[source, java]
+----
+@Outgoing("name") 
+Publisher<Message<O>> method()
+----
+| None
+| 
+
+|
+[source, java]
+----
+@Outgoing("channel") 
+Publisher<O> method()
+----
+| None
+| 
+
+|
+[source, java]
+----
+@Outgoing("channel") 
+PublisherBuilder<Message<O>> method()
+----
+| None
+| 
+
+|
+[source, java]
+----
+@Outgoing("channel") 
+PublisherBuilder<O> method()
+----
+| None
+| 
+
+|
+[source, java]
+----
+@Outgoing("channel") 
+Message<O> method()
+----
+| None
+| 
 
-Messages may be wrapped in a `Message`, which can be used to supply metadata, and also allows messages to be acknowledged. The contract for acknowledging messages is anything that accepts a `Message` is required to acknowledge it. So, if the application receives an incoming message wrapped in `Message`, it is responsible for invoking `Message.ack()`, and if the application publish an outgoing message wrapped in `Message`, then the spec implementation is responsible for invoking `Message.ack()`.
+|
+[source, java]
+----
+@Outgoing("channel")
+O method()
+----
+| None
+| 
+
+|
+[source, java]
+----
+@Outgoing("channel")
+CompletionStage<Message<O>> method()
+----
+| None
+| 
+
+|
+[source, java]
+----
+@Outgoing("channel") 
+CompletionStage<O> method()
+----
+| None
+| 
+
+|
+[source,java]
+----
+@Incoming("channel")
+Subscriber<Message<I>> method()
+----
+| Post-Processing	
+| None, Pre-Processing, Post-Processing, Manual
+
+|
+[source,java]
+----
+@Incoming("channel")
+Subscriber<I> method()
+----
+| Post-Processing	
+| None, Pre-Processing, Post-Processing
+
+|
+[source,java]
+----
+@Incoming("channel") 
+SubscriberBuilder<Message<I>> method()
+----
+| Post-Processing	
+| None, Pre-Processing, Post-Processing, Manual
+
+
+|
+[source,java]
+----
+@Incoming("channel")
+SubscriberBuilder<I> method()
+----
+| Post-Processing	
+| None, Pre-Processing, Post-Processing
+
+|
+[source,java]
+----
+@Incoming("channel")
+void method(I payload)
+----
+| Post-Processing	
+| None, Pre-Processing, Post-Processing
+
+|
+[source,java]
+----
+@Incoming("channel")
+CompletionStage<?> method(Message<I> msg)
+----
+| Post-Processing	
+| None, Pre-Processing, Post-Processing, Manual
+
+|
+[source,java]
+----
+@Incoming("channel") 
+CompletionStage<?> method(I payload)
+----
+| Post-Processing	
+| None, Pre-Processing, Post-Processing
+
+| 
+[source,java]
+----
+@Incoming("in")
+@Outgoing("out")
+Processor<Message<I>, Message<O>> method()
+----
+| Pre-Processing
+| None, Pre-Processing, Manual
+Post-Processing can be optionaly supported by implementations, however it requires a 1:1 mapping between the incoming message and the outgoing message.
+
+| 
+[source,java]
+----
+@Incoming("in")
+@Outgoing("out") 
+Processor<I, O> method();
+----
+| Pre-Processing
+| None, Pre-Processing
+Post-Processing can be optionaly supported by implementations, however it requires a 1:1 mapping between the incoming message and the outgoing message.
```

Comment:
The RS spec refers to these as "elements signaled by the Subscriber/Publisher" I wonder if we should avoid using the word "message" for things that are not "Message<O>" type?

Reply:
I've switched to element.